examples/sampling_c3/franka_c3_controller.cc

```diff
@@ -0,0 +1,908 @@
+        SortedPair(contact_geoms["EE"], contact_geoms["box"]));
+
+    //   Creating a list of contact pairs for the jack and the ground
+        SortedPair<GeometryId> ground_contact_1{
+        SortedPair(contact_geoms["corner_xynz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_2{
+        SortedPair(contact_geoms["corner_xnynz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_3{
+        SortedPair(contact_geoms["corner_nxynz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_4{
+        SortedPair(contact_geoms["corner_nxnynz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_5{
+        SortedPair(contact_geoms["corner_xyz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_6{
+        SortedPair(contact_geoms["corner_xnyz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_7{
+        SortedPair(contact_geoms["corner_nxyz"], contact_geoms["GROUND"])};
+        SortedPair<GeometryId> ground_contact_8{
+        SortedPair(contact_geoms["corner_nxnyz"], contact_geoms["GROUND"])};
+
+    // std::vector<std::vector<SortedPair<GeometryId>>>
+    //     contact_pairs;  // Exact list depends on c3_options.num_contacts_index,
+                        // but e.g. num_contacts_index = 0 means this will have
+                        // [[(ee,cap1), (ee,cap2), (ee_cap3)],
+                        //  [(ground,cap1sph1), (ground,cap1sph2),
+                        //   (ground,cap2sph1), (ground,cap2sph2),
+                        //   (ground,cap3sph1), (ground,cap3sph2)]]
+                        // and the first list (of 3) will get resolved to a single
+                        // ee-jack contact, and the second list (of 6) will get
+                        // resolved to 3 ground-jack contacts.
+    contact_pairs.push_back(ee_contact_pairs);
+
+    if(sampling_c3_options.num_contacts_index == 2 || sampling_c3_options.num_contacts_index == 3){
+        // If num_contacts_index is 2 or 3, we add an additional contact pair 
+        // between the end effector and the ground.
+        std::vector<SortedPair<GeometryId>> ee_ground_contact{
+        SortedPair(contact_geoms["EE"], contact_geoms["GROUND"])};
+        contact_pairs.push_back(ee_ground_contact);
+    }
+    std::vector<SortedPair<GeometryId>> ground_object_contact_pairs;
+    ground_object_contact_pairs.push_back(ground_contact_1);
+    ground_object_contact_pairs.push_back(ground_contact_2);
+    ground_object_contact_pairs.push_back(ground_contact_3);
+    ground_object_contact_pairs.push_back(ground_contact_4);
+    ground_object_contact_pairs.push_back(ground_contact_5);
+    ground_object_contact_pairs.push_back(ground_contact_6);
+    ground_object_contact_pairs.push_back(ground_contact_7);
+    ground_object_contact_pairs.push_back(ground_contact_8);
+    contact_pairs.push_back(ground_object_contact_pairs);
+}
```

Comment:
None of this belongs in the jack PR. FWIW many of these comments are also wrong because it looks like they were copied from the jack.